ledger-service/http-json/src/main/scala/com/digitalasset/http/json/ResponseFormats.scala

```diff
@@ -4,74 +4,79 @@
 package com.daml.http.json
 
 import akka.NotUsed
-import akka.http.scaladsl.model._
-import akka.stream.scaladsl.{Concat, Source, _}
-import akka.stream.{FanOutShape2, SourceShape, UniformFanInShape}
+import akka.http.scaladsl.model.StatusCode
+import akka.http.scaladsl.model.StatusCodes.{InternalServerError, OK}
+import akka.stream.scaladsl.{Concat, Source}
 import akka.util.ByteString
-import com.daml.fetchcontracts.util.AkkaStreamsDoobie
 import scalaz.syntax.show._
-import scalaz.{Show, \/}
+import scalaz.{Show, -\/, \/, \/-}
 import spray.json.DefaultJsonProtocol._
 import spray.json._
 
 private[http] object ResponseFormats {
   def errorsJsObject(status: StatusCode, es: String*): JsObject = {
     val errors = es.toJson
-    JsObject(statusField(status), ("errors", errors))
+    JsObject(("status", JsNumber(status.intValue)), ("errors", errors))
   }
 
   def resultJsObject[A: JsonWriter](a: A): JsObject = {
     resultJsObject(a.toJson)
   }
 
   def resultJsObject(a: JsValue): JsObject = {
-    JsObject(statusField(StatusCodes.OK), ("result", a))
+    JsObject(("status", JsNumber(OK.intValue)), ("result", a))
   }
 
   def resultJsObject[E: Show](
       jsVals: Source[E \/ JsValue, NotUsed],
       warnings: Option[JsValue],
   ): Source[ByteString, NotUsed] = {
-
-    val graph = GraphDSL.create() { implicit b =>
-      import GraphDSL.Implicits._
-
-      val partition: FanOutShape2[E \/ JsValue, E, JsValue] = b add AkkaStreamsDoobie.partition
-      val concat: UniformFanInShape[ByteString, ByteString] = b add Concat(3)
-
-      // first produce optional warnings and result element
-      warnings match {
-        case Some(x) =>
-          Source.single(ByteString(s"""{"warnings":${x.compactPrint},"result":[""")) ~> concat.in(0)
-        case None =>
-          Source.single(ByteString("""{"result":[""")) ~> concat.in(0)
+    jsVals
```

Comment:
We have to look at every item in jsVals to determine whether the payload of the response should be errors or results. To support this we change from the hardcoded `Concat(3)` in the GraphDSL and instead first collect all relevant items, before streaming the appropriate JSON object out. Note that we still need to stream it, rather than returning a single ByteString, in order to support large result sets without overwhelming the HTTP response chunk size.